test/bidirectional_graph_test.rb

```diff
@@ -31,10 +20,12 @@ def setup
     @edges.each do |(src, target)|
       @dg.add_edge(src, target)
     end
+    @eg = BidirectionalAdjacencyGraph.new
+    @gfa = BidirectionalAdjacencyGraph[1, 2, 3, 4]
   end
 
   def test_empty_graph
-    dg = BidirectionalAdjacencyGraph.new
+    dg = @eg.clone
```

Comment:
Why do we need cloning? The instance variables are created fresh in each test. Simply assigment `dg = @eg` should also work.

Reply:
Sorry. A lot going on here and my Ruby is rusty. I'll take a more careful look tomorrow.

Reply:
You're right; cloning is unnecessary. Removed throughout.